.github/workflows/test.yml

```diff
@@ -0,0 +1,33 @@
+name: test
+
+on: [push, pull_request]
+
+jobs:
+  pytest:
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      matrix:
+        os: [ubuntu]
+        python-version: ["3.9", "3.10", "3.11"]
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/pyproject.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
```

Comment:
I just stole the [workflow from spin](https://github.com/scientific-python/spin/blob/f43a217af4a39d0383738c60197f2b802d6871aa/.github/workflows/lint.yml#L24) and modified it to fit. :sweat_smile: 

Looking at [`actions/cache`'s documentation](https://github.com/actions/cache#inputs) it looks like it might be useful to keep it around.

Reply:
*those* people :roll_eyes: :joy: